### PR TITLE
Port Virgo Orbital Research Establishment code.

### PR DESCRIPTION
**Pull request by tigercat2000**
This upgrades our "mob eat mob" system to the standards of Virgo Orbital
Research Establishment.

Features:
 - Massively expanded system for mobs eating mobs
 - Mobs will no longer suffocate to death when eaten by other mobs.
 Fixes issues with xenomorphs accidentaling their hosts. Currently, the
 temperature is constant, but it can easily be variable to the mob's
 body temperature.
 - Added sounds to the process! Nothing quite as immersion breaking as
 eating someone and not even hearing a "ding your dinner is ready", so
 there is now proper swallowing sounds and more!
 - Refactored mob breathing code. handle_internal_lifeform() has been
 expanded to all movable atoms, instead of just objects, allowing for
 custom environments inside of mobs.
 - You may now resist to attempt to escape whatever ate you. You
 probably won't succeed, as it has a 0% chance by default.

Okay, okay, I know what you all want: Get to the point Uncle Tiger, tell us
what the first bullet point means!

I'll do just that! A quick summary of the changes:
Mobs inside mobs are now abstracted to a /datum/belly system, which
controls everything about the mob's environment and effects on the mob.
This effecitvely means that we are now able to provide a larger range of
functionality for the mob eat mob system, including, but not limited to:
 - You may now have up to 25 different "destinations". These are all
 customizable through the VORE (Very Orbital Repeated Emaciation) tab of your command lists.
   - Specifically, you may customize the following properties:
     - The name of the "destination" which is shown to anyone who ends up
     inside it using the VORE panel.
     - A flavor text for the location. Limit of 1024 characters.
     - One sound out of 12 available which is played when a mob is
     ingested.
     - Messages produced when a mob struggles inside you, both for the
     mob and anyone within one tile of you. (Seperate messagesets, but you
     can change both)
     - Messages produced when a mob is digested, both for the mob and
     anyone within one tile of you.
     - Messages produced for anyone who examines you when you have a mob
     in the location.
 - The VORE Panel also allows you to set the taste of your flesh, which
 is given to anything that eats you, as well as anyone who licks you-
 another new feature.
   - Last part of this: the VORE panel allows you to disable being
   digested. You're still fucked though, since you're trapped, so why
   bother?
 - Mobs may now struggle inside of you, producing a message and a noise.
   You can choose what this does- refer to the following. Please note that
   all of these are combinable (except for obvious conflicts)
   - Print message, sound, no ability to escape (conflicts with everything else)
   - Set a percentage chance for the mob to escape from your belly,
   which is rolled every time they resist
   - Set a percentage chance for resisting to cause them to slide into a
   different "destination" of your body. Where they end up is also
   customizable.
   - Set a percentage chance for "absorb" mode to be enabled. More on
   this and the next bullet point later.
   - Set a percentage chance for "digest" mode to be enabled.
 - Eating is no longer limited to purely "Mob grabs mob 2, mob eats mob 2".
 There are now three distinctive modes of eating:
   - Mob 1 (now refered to as "Hungry") grabs Mob 2 (now refered to as
   "Fucked"), and clicks on "Hungry" in order to consume "Fucked".
   - "Hungry" is grabbed by "Fucked", and "Fucked" clicks on "Hungry" to
   feed themselves to "Hungry".
   - "Fucked" is grabbed by "Hungry", and "Hungry" clicks on a new third
   mob labeled "Innocent bystander" to feed "Fucked" to "Innocent
   bystander".
 - You now have a choice of 10 different "modes" of operation, seperated
 by location.
   - "Hold": Does exactly what it sounds like. Anything within this
   organ remains trapped there and otherwise perfectly unharmed.
   - "Digest": Starts digesting any mobs within it, so long as they are
   not marked as indigestible. This does both brute and fire damage to
   the mobs, and once they die, their bodies are qdeleted and your
   custom digestion messages are triggered. Average time from full
   health to death for an unarmored victim is about 3-4 minutes. You
   gain a significant amount of nutrition from this, and they lose
   exactly what you take until they are dead.
   - "Heal": Opposite of digestion. It heals them, spending your
   nutrition.
   - "Absorb": Like digestion, but instead of damage and qdeletion, it
   simply drains them of nutrition and then absorbs them into your
   organ. They can no longer do anything at this point, as they are
   constantly Weaken()ed by your belly.
   - "Drain": Drains the person inside you of nutrition and gives it to
   you. When they run out, it stops draining them, and they're just
   trapped as if it was in Hold mode.
   - "Unabsorb": Undoes the effects of "Absorb" mode. You must have 100
   or more nutrition to do this, which shouldn't be a problem given that
   absorbing someone generally gives you upwards of 200.
   - "Transform - Male": This mode transforms the victim to look like
   you, except it always makes them male as well. For the ladies out
   there, you can make the man you always wanted: You, but with a cock!
   - "Transform - Female": Opposite of "Transform - Male". They become a
   clone of you, but female. Guys? ;)
   - "Transform - Keep Gender": This one makes them look like you as
   closely as it can, but they do not change gender. The wise option,
   for when you kidnap a victim you actually like the gender of!
   - "Egg": It makes you shit out an egg with them inside of it. They
   can break out at any time.

Technical details:
 - Almost all of this is implemented using a new proc called attempt_vr.
 This means that the majority of things that are touched outside of the
 vore folder only have one line added to them, which is failproof.

### CODE REVIEW

**Comment by tylerh95** [comment]
this is why our github needs to be burned

**Comment by SigholtStarsong** [comment]
Cleanse. Purge. Kill.

**Comment by Xantholne** [comment]
Automatic merge pls

File: code/_globalvars/lists/global_list_vr.dm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		"Squish4" = 'sound/vore/squish4.ogg')
+
+
+var/global/list/global_egg_types = list(
+		"Unathi" 		= UNATHI_EGG,
+		"Tajaran" 		= TAJARAN_EGG,
+		"Akula" 		= AKULA_EGG,
+		"Skrell" 		= SKRELL_EGG,
+		"Sergal" 		= SERGAL_EGG,
+		"Human"			= HUMAN_EGG,
```
Comment (SamHPurp) [inline code/_globalvars/lists/global_list_vr.dm]: *scream

**Comment by SheepFighter** [comment]
Move over Lavaland!

File: code/_globalvars/lists/global_list_vr.dm (hunk 1 above)
Comment (SheepFighter) [inline code/_globalvars/lists/global_list_vr.dm]: @jughu Will be most pleased.

**Comment by DesolateG** [comment]
I like this.. let's merge asap.

**Comment by SomanBat** [comment]
Speedy merge this is required to bring Para RP up to standard.

**Comment by SigholtStarsong** [comment]
@tigercat2000 Will this re-merge dildoes?

**Comment by tigercat2000** [comment]
@SigholtStarsong I'll put up a second PR for that when this is merged 😏

**Comment by monster860** [comment]
I love how the PR number is 6969.

Did you intentionally wait for this opportunity?

**Comment by Anticept** [comment]
I don't mind the mob eating mob thing updated, as long as it doesn't get weird. Getting weird should fall under the ERP rule.

**Comment by MarsM0nd** [comment]
Well, if this removes the ability to stun/robust a xeno that has eaten you, i would say it's a good change.

**Comment by SamHPurp** [comment]
@monster860 of course

**Comment by Nickvr628** [comment]
Merge this at once.

**Comment by Aracino** [comment]
I don't see how this could possibly go wrong!

**Comment by mrturkeytoe** [comment]
Can we port their really neat electronics system they got from Polaris too?

**Comment by taukausanake** [comment]
One must wonder just how much work goes into doing PRs like this and if they actually work. I mean just imagine how much good could be done if this much effort was put into other parts of the code implying the above.
I thought we were porting something actually new and cool. Then I Googled "Virgo Orbital Research Establishment" and found out it was an acronym. Then I saw the PR#. Disappointed

**Comment by mrturkeytoe** [comment]
Chill man, this is like a ritual for tiger.

**Comment by DesolateG** [comment]
@taukausanake what can be more cool than eating someone, making them look female, then laying them as an egg?

**Comment by granodd** [comment]
Y'know, when I first started reading this, I was like "Eh, this is pretty weird, but if you wanna improve xenos or something and give them a gulp sound, whatever mate that's fine". 

But then I read "Vore tab" and "lick/taste system" and immediately changed that feeling to "Oh..."

**Comment by tigercat2000** [comment]
@taukausanake Of course it works. What's the point of making a sick meme PR if you don't actually put any effort into it? This is perfectly mergable, and would work exactly as desired- that's the fun of it!

**Comment by OrbisAnima** [comment]
So, this is Paradise April Fools PR? heh, april 1 will be fun

**Comment by KasparoVy** [comment]
What a God.

**Comment by Arokha** [comment]
😄

**Pull request closed without merging**